evm/src/fuzz/strategies/state.rs

```diff
@@ -10,18 +10,51 @@ use ethers::{
     types::{Address, Log, H256, U256},
 };
 use foundry_common::contracts::{ContractsByAddress, ContractsByArtifact};
+use lru::LruCache;
 use parking_lot::RwLock;
 use proptest::prelude::{BoxedStrategy, Strategy};
 use revm::{
     db::{CacheDB, DatabaseRef},
     opcode, spec_opcode_gas, Filth, SpecId,
 };
-use std::{collections::BTreeSet, io::Write, sync::Arc};
+use std::{
+    collections::BTreeSet,
+    io::Write,
+    ops::{Deref, DerefMut},
+    sync::Arc,
+};
 
 /// A set of arbitrary 32 byte data from the VM used to generate values for the strategy.
 ///
 /// Wrapped in a shareable container.
-pub type EvmFuzzState = Arc<RwLock<BTreeSet<[u8; 32]>>>;
+pub type EvmFuzzState = Arc<RwLock<FuzzDictionary>>;
+
+#[derive(Debug)]
+pub struct FuzzDictionary {
+    inner: BTreeSet<[u8; 32]>,
+    /// Addresses that already had their PUSH bytes collected.
+    cache: LruCache<Address, u8>,
```

Comment:
this is just a simple map with very small types.

I think we should just make this a regular HashMap instead of a cache, or are they any sideffects with caching everything?

Reply:
this was my recommendation but yeah seems like this shouldn't require a full LRU cache given that it won't be THAT huge. i guess lru caches are better for long running daemon like processes, so shouldnt be required here.